check_logs.py

```diff
@@ -252,30 +261,47 @@ def get_errors_from_ansible_log(args, log_url):
     logging.debug("Getting errors from ansible log [%s]", log_url)
 
     # Extracts the system role name
-    extracted_part = log_url.split("logs/")[1]
-    start_index = extracted_part.find("tf_") + 3  # +3 to skip "tf_"
-    if start_index > 2:
-        end_index = extracted_part.index("-", start_index)
-        role = extracted_part[start_index:end_index]
-        pattern = r"ANSIBLE-(\d+\.\d+)"
-        ansible_version_matches = re.findall(pattern, log_url)
-        ansible_version = (
-            ansible_version_matches[0] if ansible_version_matches else "UNKNOWN"
-        )
+    if "logs/tf_" in log_url:
+        extracted_part = log_url.split("logs/")[1]
+        start_index = extracted_part.find("tf_") + 3  # +3 to skip "tf_"
+        if start_index > 2:
+            end_index = extracted_part.index("-", start_index)
+            role = extracted_part[start_index:end_index]
+            pattern = r"ANSIBLE-(\d+\.\d+)"
+            ansible_version_matches = re.findall(pattern, log_url)
+            ansible_version = (
+                ansible_version_matches[0] if ansible_version_matches else "UNKNOWN"
+            )
     else:
         # https://....//SYSTEM-ROLE-$ROLENAME_$TEST_NAME.yml-legacy-ANSIBLE-2.log
         match = SYSTEM_ROLE_LOG_RE.search(log_url)
-        role = match.group("role")
-        if args.role != ["ALL"] and role not in args.role:
-            logging.info(
-                "Skipping log - role [%s] not in args.role [%s]: [%s]",
-                role,
-                str(args.role),
-                log_url,
-            )
-            return []
-        # test = match.group(2)  # unused for now
-        ansible_version = match.group("ansible_ver")
+        if match:
+            role = match.group("role")
+            if args.role != ["ALL"] and role not in args.role:
+                logging.info(
+                    "Skipping log - role [%s] not in args.role [%s]: [%s]",
+                    role,
+                    str(args.role),
+                    log_url,
+                )
+                return []
+            # test = match.group(2)  # unused for now
+            ansible_version = match.group("ansible_ver")
+        else:
+            # testing farm - https://...../data/$ROLENAME-$TESTNAME-ANSIBLE-$VER-$TFTESTNAME-$STATUS.log
+            match = SYSTEM_ROLE_TF_LOG_RE.search(log_url)
```

Comment:
Note that David suggested removing ROLE_NAME completely to make it work like upstream_testsuite. So that when `SYSTEM_ROLES_ONLY_TESTS` is not provided than we test all roles.
But that's in planning stage, we can revise this later.